src/devices/bus/hp_dio/hp98603.cpp

```diff
@@ -0,0 +1,74 @@
+// license:BSD-3-Clause
+// copyright-holders:Sven Schnelle
+/***************************************************************************
+
+  HP98603 BASIC ROM card
+
+***************************************************************************/
+
+#include "emu.h"
+#include "hp98603.h"
+#include "screen.h"
+#include "softlist.h"
+#include "bus/generic/slot.h"
+#include "bus/generic/carts.h"
+
+DEFINE_DEVICE_TYPE(HPDIO_98603, dio16_98603_device, "dio98603", "HP98603 BASIC ROM card")
+
+MACHINE_CONFIG_MEMBER( dio16_98603_device::device_add_mconfig )
+        MCFG_GENERIC_CARTSLOT_ADD("cartslot", generic_plain_slot, "98603_cart")
+
+        MCFG_GENERIC_EXTENSIONS("bin,rom")
+        MCFG_GENERIC_LOAD(dio16_98603_device, hp98603_rom_slot)
+	MCFG_SOFTWARE_LIST_ADD("cart_list", "98603")
+MACHINE_CONFIG_END
+
+const tiny_rom_entry *dio16_98603_device::device_rom_region() const
+{
+	return NULL;
+}
+
+DEVICE_IMAGE_LOAD_MEMBER(dio16_98603_device, hp98603_rom_slot)
+{
+        uint32_t size = m_romslot->common_get_size("rom");
+        m_romslot->rom_alloc(size, GENERIC_ROM16_WIDTH, ENDIANNESS_BIG);
+        m_romslot->common_load_rom(m_romslot->get_rom_base(), size, "rom");
+        return image_init_result::PASS;
+
+}
+
+dio16_98603_device::dio16_98603_device(const machine_config &mconfig, const char *tag, device_t *owner, uint32_t clock) :
+	dio16_98603_device(mconfig, HPDIO_98603, tag, owner, clock)
+{
+}
+
+dio16_98603_device::dio16_98603_device(const machine_config &mconfig, device_type type, const char *tag, device_t *owner, uint32_t clock) :
+	device_t(mconfig, type, tag, owner, clock),
+	device_dio16_card_interface(mconfig, *this),
+	m_romslot(*this, "cartslot")
+{
+}
+
+void dio16_98603_device::device_start()
+{
+	set_dio_device();
+}
+
+void dio16_98603_device::device_reset()
+{
+	if (m_romslot->exists()) {
+		m_dio->install_memory(0x100000, 0x1fffff, read16_delegate(FUNC(dio16_98603_device::rom_r), this),
+				      write16_delegate(FUNC(dio16_98603_device::rom_w), this));
+	}
+}
+
+READ16_MEMBER(dio16_98603_device::rom_r)
+{
+	uint16_t data = m_romslot->read_rom(space, offset*2, mem_mask) | \
```

Comment:
C++ does not need the backslash preprocessor line continuation directive in cases like this - it interprets the lack of a semicolon as an indication that the expression will continue on the next line.  Could you please not use unnecessary backslash line continuation directives like this?